src/splat-convert.ts

```diff
@@ -51,16 +48,16 @@ const convertPly = (splatData: GSplatData, modelMat: Mat4) => {
     const v = new Vec3();
     const q = new Quat();
 
-    const x_off = props.indexOf('x') * 4;
-    const y_off = props.indexOf('y') * 4;
-    const z_off = props.indexOf('z') * 4;
-    const r0_off = props.indexOf('rot_0') * 4;
-    const r1_off = props.indexOf('rot_1') * 4;
-    const r2_off = props.indexOf('rot_2') * 4;
-    const r3_off = props.indexOf('rot_3') * 4;
-    const scale0_off = props.indexOf('scale_0') * 4;
-    const scale1_off = props.indexOf('scale_1') * 4;
-    const scale2_off = props.indexOf('scale_2') * 4;
+    const x_off = props.findIndex((p: any) => p.name === 'x') * 4;
+    const y_off = props.findIndex((p: any) => p.name === 'y') * 4;
+    const z_off = props.findIndex((p: any) => p.name === 'z') * 4;
+    const r0_off = props.findIndex((p: any) => p.name === 'rot_0') * 4;
+    const r1_off = props.findIndex((p: any) => p.name === 'rot_1') * 4;
+    const r2_off = props.findIndex((p: any) => p.name === 'rot_2') * 4;
+    const r3_off = props.findIndex((p: any) => p.name === 'rot_3') * 4;
+    const scale0_off = props.findIndex((p: any) => p.name === 'scale_0') * 4;
+    const scale1_off = props.findIndex((p: any) => p.name === 'scale_1') * 4;
+    const scale2_off = props.findIndex((p: any) => p.name === 'scale_2') * 4;
```

Comment:
How about simplifying this slightly? (untested)

```suggestion
    const propIndex = (name: string) => props.findIndex((p: any) => p.name === name);
    const x_off = propIndex('x') * 4;
    const y_off = propIndex('y') * 4;
    const z_off = propIndex('z') * 4;
    const r0_off = propIndex('rot_0') * 4;
    const r1_off = propIndex('rot_1') * 4;
    const r2_off = propIndex('rot_2') * 4;
    const r3_off = propIndex('rot_3') * 4;
    const scale0_off = propIndex('scale_0') * 4;
    const scale1_off = propIndex('scale_1') * 4;
    const scale2_off = propIndex('scale_2') * 4;
```